installers/AppVeyor.hs

```diff
@@ -0,0 +1,160 @@
+buildJobs :: Monoid r => Getting r Value Job
+buildJobs = key "build" . key "jobs" . _Array . traverse . _JSON
+
+-- | Looks into AppVeyor API job artifacts json to get the filenames
+artifactFilenames :: Monoid r => Getting r Value Text
+artifactFilenames = _Array . traverse . key "fileName" . _String
+
+-- | The AppVeyor API URL of an build, given its web URL.
+buildAPI :: GH.URL -> Maybe URI
+buildAPI = parseAbsoluteURI . T.unpack . T.replace "/project/" "/api/projects/" . getUrl
+
+-- | The AppVeyor API URL to list artifacts belonging to a job.
+jobArtifactsURI :: Text -> String
+jobArtifactsURI jobId = "https://ci.appveyor.com/api/buildJobs/" <> T.unpack jobId <> "/artifacts"
+
+-- | Use GitHub API to find AppVeyor build status
+appVeyorURL :: CardanoSource -> IO GH.URL
+appVeyorURL src = fmap collect <$> statusFor' src >>= \case
+  Right (Just u) -> pure u
+  Right Nothing  -> throwM $ StatusMissingError src
+  Left err       -> throwM $ GitHubStatusError src err
+  where
+    collect = head . mapMaybe statusTargetUrl . filter isAppVeyor .
+              toList . combinedStatusStatuses
+    isAppVeyor st = statusContext st == Just "continuous-integration/appveyor/branch"
+
+statusFor' :: CardanoSource -> IO (Either GH.Error GH.CombinedStatus)
+statusFor' CardanoSource{..} = statusFor noAuth srcOwner srcRepo srcRev
+  where noAuth = BasicAuth "" ""
+
+data JobStatus = JobSuccess | JobFailed | JobRunning
+               | JobUnknown Text deriving (Show, Eq, Generic)
+
+data Job = Job { jobId :: Text
+               , jobStatus :: JobStatus
+               } deriving (Show, Eq, Generic)
+
+instance FromJSON JobStatus where
+  parseJSON = withText "status" $ \case
+    "success" -> pure JobSuccess
+    "failed" -> pure JobFailed
+    "running" -> pure JobRunning
+    other -> pure $ JobUnknown other
+
+instance FromJSON Job where
+  parseJSON = withObject "Job" $ \o -> Job <$> o .: "jobId" <*> o .: "status"
+
+instance ToJSON JobStatus
+instance ToJSON Job -- Only required for _JSON prism
+
+----------------------------------------------------------------------------
+
+-- | Load commit hash and GitHub repo name from cardano-sl-src.json.
+getCardanoRev :: FilePath -> IO CardanoSource
+getCardanoRev src = eitherDecode <$> L8.readFile src >>= \case
+  Right rev -> pure rev
+  Left err -> throwM (SourceJSONDecodeError err)
+
+data CardanoSource = CardanoSource
+  { srcOwner :: Name Owner
+  , srcRepo  :: Name Repo
+  , srcRev   :: Name Commit
+  } deriving (Show, Eq)
+
+instance FromJSON CardanoSource where
+  parseJSON = withObject "fetchGit source" $ \o -> do
+    (owner, repo) <- parseGitHubURL <$> o .: "url"
+    CardanoSource owner repo <$> o .: "rev"
+
+instance FromJSON URI where
+  parseJSON = withText "Absolute URI" $ \u -> case parseAbsoluteURI (T.unpack u) of
+    Just uri -> pure uri
+    Nothing -> fail "Could not parse absolute URI"
+
+-- | Gets owner/repo from gitcom.com URL
```

Comment:
Seeing how we have `*parsec` in our transitive deps, maybe we could be somewhat defensive and define a proper `Parser` for this?

Again, just a thought to consider.

Reply:
Why?

Reply:
Being defensive, as I said.